### PR TITLE
Back out "Back out "[aotinductor] Add example_value metadata to nodes (#112415)""

### PR DESCRIPTION
**Pull request by frank-wei**
Reviewed By: khabinov

Differential Revision: D51223990

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler @avikchaudhuri @gmagogsfm @zhxchen17 @tugsbayasgalan @angelayi @suo @ydwu4

### CODE REVIEW

**Review by zhxchen17 — CHANGES_REQUESTED** [review]
I'm not sure why we need to run fake prop here. Could you explain?

**Comment by frank-wei** [comment]
> I'm not sure why we need to run fake prop here. Could you explain?

Some of the following passes need to use certain tensor's meta["example_value"].shape in their condition check.

**Review by angelayi — APPROVED** [review]

**Review by zhxchen17 — COMMENTED** [review]
Yeah the contract makes sense but I still have some question with the code placement.

File: torch/_export/__init__.py [hunk 1]
```diff
@@ -955,6 +955,10 @@ def aot_compile(
     # We want to export to Torch IR here to utilize the pre_grad passes in
     # inductor, which run on Torch IR.
     gm = _export_to_torch_ir(f, args, kwargs, constraints)
+
+    from torch._export.passes.fake_tensor_prop import FakeTensorProp
+    FakeTensorProp(gm).run()
```
[the remark this answers is not in the document]
Comment (frank-wei) [reply torch/_export/__init__.py]: Moved it.

**Review by zhxchen17 — APPROVED** [review]
thanks for addressing my questions!

**Pull request closed without merging**